### PR TITLE
[SW-320] Unblock H2OConf test

### PR DESCRIPTION
**Pull request by jakubhava**
It's workaround on the Gradle level

### CODE REVIEW

File: py/build.gradle [hunk 1]
```diff
@@ -170,27 +185,24 @@ artifacts {
...
-
+    preparePythonTestEnv(environment)
+    commandLine getOsSpecificCommandLine([pythonexec, "tests/tests_unit_conf.py"])
+}
 
+task testPythonConversions(type: Exec, dependsOn: distPython){
+    doFirst {
+        preparePythonEnv(environment)
+    }
+    preparePythonTestEnv(environment)
```
Comment (mmalohlava) [inline py/build.gradle]: Why one preparation is at configuration stage and second one at execution?

File: py/tests/tests_unit_conf.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        cls._hc = H2OContext.getOrCreate(cls._spark, h2o_conf)
+
+    @classmethod
+    def tearDownClass(cls):
+        h2o.cluster().shutdown()
+        test_utils.tear_down_class(cls)
+
+    # test passing h2o_conf to H2OContext
+    def test_h2o_conf(self):
+        self.assertEquals(self._hc.get_conf().cloud_name(),"test-cloud", "Configuration property cloud_name should match")
```
Comment (mmalohlava) [inline py/tests/tests_unit_conf.py]: is it possible to test the rest properties of config? Futhermore, the test should fail if we add a new config property.

File: py/build.gradle (hunk 1 above)
Comment (jakubhava) [reply py/build.gradle]: The `preparePythonTestEnv(environment)` depends on the JAR archive of SW, that's why it is in the execution phase

**Comment by jakubhava** [comment]
The original problem was that when having multiple test classes in a single python file, the second tests were blocking the gradle as the underlying JVM was shared and H2O can't be restarted. By moving the H2OConf test into a new file and executing it separately we create a workaround for this issue as the tests are executed independently

File: py/tests/tests_unit_conf.py (hunk 2 above)
Comment (jakubhava) [reply py/tests/tests_unit_conf.py]: I created these 2 PRs ( just to keep this one simple ) https://0xdata.atlassian.net/browse/SW-578 - for testing all properties

https://0xdata.atlassian.net/browse/SW-579 - for testing consistency of properties.

We might actually automatically generate the python configuration based on the Scala one.

**Pull request merged**